src/autopas/fastMultipoleMethod/FmmTreeNode.h

```diff
@@ -0,0 +1,211 @@
+#include "autopas/utils/ArrayMath.h"
+
+namespace autopas::fmm {
+
+// forward declaration
+class FmmTree;
+
+class FmmTreeNode {
+ public:
+  std::string name;
+  FmmTreeNode(FmmTreeNode *parent, std::array<double, 3> boxMin, std::array<double, 3> boxMax)
+      : parent(parent),
+        boxMin(boxMin),
+        boxMax(boxMax),
+        boxCenter(autopas::utils::ArrayMath::mulScalar(autopas::utils::ArrayMath::add(boxMin, boxMax), 0.5)),
+        sphereRadius(autopas::utils::ArrayMath::L2Norm(autopas::utils::ArrayMath::sub(boxMax, boxCenter))) {
+    if (parent == nullptr) {
+      depth = 0;
+    } else {
+      depth = parent->depth + 1;
+    }
+
+    // std::cout << "sphereRadius = " << sphereRadius << std::endl;
+  }
+  FmmTreeNode(const FmmTreeNode &) = delete;
+  FmmTreeNode &operator=(const FmmTreeNode &) = delete;
+  FmmTreeNode(FmmTreeNode &&) = default;
+  FmmTreeNode &operator=(FmmTreeNode &&) = default;
+
+  void split(std::array<double, 3> firstBoxMax, std::array<double, 3> secondBoxMin) {
+    if (children.empty()) {
+      children.emplace_back(this, boxMin, firstBoxMax);
+      children.emplace_back(this, secondBoxMin, boxMax);
+    } else {
+      autopas::utils::ExceptionHandler::exception("trying to split an already split FmmTreeNode");
+    }
+  }
+
+  void insertLeaves(FmmTreeNode &otherNode) {
+    if (otherNode.isLeaf()) {
+      if (ancestorInteractionList.find(&otherNode) == ancestorInteractionList.end()) {
+        interactionList.insert(&otherNode);
+      }
+    } else {
+      insertLeaves(otherNode.getChild(0));
+      insertLeaves(otherNode.getChild(1));
+    }
+  }
+
+  bool checkInteract(FmmTreeNode &otherNode) {
+    /*std::cout << "checkInteract:" << std::endl;
+    std::cout << name << std::endl;
+    std::cout << otherNode.name << std::endl;*/
+    double r1 = sphereRadius;
+    double r2 = otherNode.sphereRadius;
+    double maxRadius = std::max(r1, r2);
+    double distance = autopas::utils::ArrayMath::L2Norm(autopas::utils::ArrayMath::sub(boxCenter, otherNode.boxCenter));
+    // std::cout << "maxRadius = " << maxRadius << std::endl;
+    // std::cout << "distance = " << distance << std::endl;
+    return distance > 3 * maxRadius;
+  }
+
+  int findLargestIndex(std::array<double, 3> array) {
+    int largestIndex = 0;
+    if (array[1] > array[largestIndex]) {
+      largestIndex = 1;
+    }
+    if (array[2] > array[largestIndex]) {
+      largestIndex = 2;
+    }
+    return largestIndex;
+  }
+
+  void initLists(FmmTreeNode &otherNode) {
+    if (checkInteract(otherNode)) {
+      interactionList.insert(&otherNode);
+    } else {
+      auto boxSize1 = utils::ArrayMath::sub(boxMax, boxMin);
+      auto boxSize2 = utils::ArrayMath::sub(otherNode.boxMax, otherNode.boxMin);
+
+      auto largestIndex1 = findLargestIndex(boxSize1);
+      auto largestIndex2 = findLargestIndex(boxSize2);
+
+      auto largestSize1 = boxSize1[largestIndex1];
+      auto largestSize2 = boxSize2[largestIndex2];
```

Comment:
you should probably move this into a member function, e.g, `getLargestSize()` as it makes the code a bit easier to read.